mavsdkserver stop method is not working in the last release

I was testing the android-client example, and the mavsdkServer.stop() call in the onPause of the MapsActivity works fine. I mean, I am testing with a sim_vehicle, and I can see on the console that when I connect to the vehicle I see "New connection on serial port 0" on the console, and when I leave the MapsActivity, I see "Closed connection on serial port 0".

This example is using:
- implementation 'io.mavsdk:mavsdk:0.7.0'
- implementation 'io.mavsdk:mavsdk-server:0.7.0'

I tried to update to the last version (1.1.1), and the connection to the sim_vehicle works fine, but with this last version what is not working is the stop method. When I leave tha activity, I don't see the "Closed connection on serial port 0" on the console, and I have to close the app to see this message. I am using the same code, the only thing I did was to update the version of the mavsdk and the mavsdk-server.

1. Is this a known bug?
2. Are you working to solve it in the following release?

Thanks!

Thats what I thought... Have you ever seen this error?

```
-- Configuring done
-- Generating done
-- Build files have been written to: /home/dronedojo/github_repos/MAVSDK/build/default/third_party/mavlink
[ 12%] Performing download step (git clone) for 'mavlink'
Cloning into 'mavlink'...
fatal: reference is not a tree: 3b52eac09c2e37325e4bc49cd2667ea37bf1d7d2
CMake Error at /home/dronedojo/github_repos/MAVSDK/build/default/third_party/mavlink/mavlink/tmp/mavlink-gitclone.cmake:49 (message):
  Failed to checkout tag: '3b52eac09c2e37325e4bc49cd2667ea37bf1d7d2'


CMakeFiles/mavlink.dir/build.make:97: recipe for target 'mavlink/src/mavlink-stamp/mavlink-download' failed
make[2]: *** [mavlink/src/mavlink-stamp/mavlink-download] Error 1
CMakeFiles/Makefile2:82: recipe for target 'CMakeFiles/mavlink.dir/all' failed
make[1]: *** [CMakeFiles/mavlink.dir/all] Error 2
Makefile:90: recipe for target 'all' failed
make: *** [all] Error 2
CMake Error at third_party/cmake/build_target.cmake:51 (message):
  /home/dronedojo/github_repos/MAVSDK/build/default/third_party/mavlink
  failed to build!
Call Stack (most recent call first):
  third_party/CMakeLists.txt:31 (build_target)


-- Configuring incomplete, errors occurred!
See also "/home/dronedojo/github_repos/MAVSDK/build/default/CMakeFiles/CMakeOutput.log".
```

Wait...

> the error is the same I posted 8 days ago

What you posted back then said:

```
Cloning into 'mavlink'...
fatal: reference is not a tree: 3b52eac09c2e37325e4bc49cd2667ea37bf1d7d2
CMake Error at /home/dronedojo/github_repos/MAVSDK/build/default/third_party/mavlink/mavlink/tmp/mavlink-gitclone.cmake:49 (message):
  Failed to checkout tag: '3b52eac09c2e37325e4bc49cd2667ea37bf1d7d2'
```

What I see in your video is:

```
ImportError: No module named 'future'
```

Am I missing something? At least you should solve that error, which is a python dependency that needs to be installed.

It's crashing for me, any idea? (I'm not very familiar with this stuff)

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: io.mavsdk.androidclient, PID: 22933
    java.lang.RuntimeException: Unable to start activity ComponentInfo{io.mavsdk.androidclient/io.mavsdk.androidclient.MapsActivity}: java.lang.SecurityException: getDataNetworkTypeForSubscriber
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3778)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3940)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:109)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2345)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:233)
        at android.os.Looper.loop(Looper.java:344)
        at android.app.ActivityThread.main(ActivityThread.java:8210)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1034)
     Caused by: java.lang.SecurityException: getDataNetworkTypeForSubscriber
        at android.os.Parcel.createExceptionOrNull(Parcel.java:2441)
        at android.os.Parcel.createException(Parcel.java:2425)
        at android.os.Parcel.readException(Parcel.java:2408)
        at android.os.Parcel.readException(Parcel.java:2350)
        at com.android.internal.telephony.ITelephony$Stub$Proxy.getNetworkTypeForSubscriber(ITelephony.java:9300)
        at android.telephony.TelephonyManager.getNetworkType(TelephonyManager.java:2990)
        at android.telephony.TelephonyManager.getNetworkType(TelephonyManager.java:2954)
        at com.mapbox.android.telemetry.TelemetryUtils.obtainCellularNetworkType(TelemetryUtils.java:165)
        at com.mapbox.mapboxsdk.module.telemetry.PhoneState.<init>(PhoneState.java:39)
        at com.mapbox.mapboxsdk.module.telemetry.TelemetryImpl.onAppUserTurnstileEvent(TelemetryImpl.java:46)
        at com.mapbox.mapboxsdk.maps.MapView.onCreate(MapView.java:303)
        at io.mavsdk.androidclient.MapsActivity.onCreate(MapsActivity.java:77)
        at android.app.Activity.performCreate(Activity.java:8129)
        at android.app.Activity.performCreate(Activity.java:8109)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1344)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3747)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3940) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:109) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2345) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loopOnce(Looper.java:233) 
        at android.os.Looper.loop(Looper.java:344) 
        at android.app.ActivityThread.main(ActivityThread.java:8210) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1034) 
```

--- a/mavsdk_server/src/main/cpp/native-lib.cpp
+++ b/mavsdk_server/src/main/cpp/native-lib.cpp
@@ -51,4 +51,11 @@ extern "C"
         auto mavsdk_server = reinterpret_cast<MavsdkServer*>(mavsdkServerHandle);
         mavsdk_server_stop(mavsdk_server);
     }
+
+    JNIEXPORT void JNICALL
+    Java_io_mavsdk_mavsdkserver_MavsdkServer_destroy(JNIEnv* env, jobject thiz, jlong mavsdkServerHandle)
+    {
+        auto mavsdk_server = reinterpret_cast<MavsdkServer*>(mavsdkServerHandle);
+        mavsdk_server_destroy(mavsdk_server);
+    }
 };

--- a/mavsdk_server/src/main/java/io/mavsdk/mavsdkserver/MavsdkServer.java
+++ b/mavsdk_server/src/main/java/io/mavsdk/mavsdkserver/MavsdkServer.java
@@ -83,4 +83,10 @@ public class MavsdkServer {
   }
 
   private native void stop(long mavsdkServerHandle);
+
+  public void destroy() {
+    destroy(mavsdkServerHandle);
+  }
+
+  private native void destroy(long mavsdkServerHandle);
 }

